File: Java/PrisonerSimulator/src/Player/IPlayerPlayFacade.java
package Player;

import Game.Decision;

public interface IPlayerPlayFacade {
    Decision makeDecision(int rivalId);

    void storeDecision(int rivalId, Decision decision);

    void updateScore(int delta, int rivalId);

    int getId();

    String getName();

    Decision getLastDecision(int rivalId);

    int getScore();

    int getLastDelta(int rivalId);
}

File: Kotlin/PrisonerSimulator/src/player/IPlayerPlayFacade.kt
package player

import game.Decision

interface IPlayerPlayFacade {
    val id: Int

    val name: String

    val score: Int

    fun makeDecision(rivalId: Int): Decision

    fun storeDecision(rivalId: Int, decision: Decision)

    fun updateScore(delta: Int, rivalId: Int)

    fun getLastDelta(rivalId: Int): Int?

    fun getLastDecision(rivalId: Int): Decision?
}
